src/client/serialize/daos_serialize.c

```diff
@@ -0,0 +1,1025 @@
+	}
+	entry = &prop->dpp_entries[4];
+	rc = deserialize_uint(file_id, entry,
+			      "DAOS_PROP_CO_CSUM_SERVER_VERIFY");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[5];
+	rc = deserialize_uint(file_id, entry, "DAOS_PROP_CO_REDUN_FAC");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[6];
+	rc = deserialize_uint(file_id, entry, "DAOS_PROP_CO_REDUN_LVL");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[7];
+	rc = deserialize_uint(file_id, entry, "DAOS_PROP_CO_SNAPSHOT_MAX");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[8];
+	rc = deserialize_uint(file_id, entry, "DAOS_PROP_CO_COMPRESS");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[9];
+	rc = deserialize_uint(file_id, entry, "DAOS_PROP_CO_ENCRYPT");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[10];
+	rc = deserialize_str(file_id, entry, "DAOS_PROP_CO_OWNER");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[11];
+	rc = deserialize_str(file_id, entry,
+				       "DAOS_PROP_CO_OWNER_GROUP");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[12];
+	rc = deserialize_uint(file_id, entry, "DAOS_PROP_CO_DEDUP");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[13];
+	rc = deserialize_uint(file_id, entry,
+					"DAOS_PROP_CO_DEDUP_THRESHOLD");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[14];
+	rc = deserialize_uint(file_id, entry,
+					"DAOS_PROP_CO_ALLOCED_OID");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	entry = &prop->dpp_entries[15];
+	/* read acl as a list of strings in deserialize, then convert back to
+	 * acl for property entry
+	 */
+	rc = deserialize_acl(file_id, entry, "DAOS_PROP_CO_ACL");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	if (deserialize_label)
+		prop->dpp_entries[16].dpe_str = strdup(label_entry->dpe_str);
+	*cont_type = prop->dpp_entries[0].dpe_val;
+	*_prop = prop;
+out:
+	/* close container after checking if label exists in pool */
+	if (close_cont) 
+		daos_cont_close(coh, NULL);
+	daos_prop_free(label);
+	return rc;
+}
+
+int
+daos_cont_deserialize_props(daos_handle_t poh, char *filename,
+			    daos_prop_t **props, uint64_t *cont_type)
+{
+	int	rc = 0;
+	hid_t	file_id = 0;
+	
+	fprintf(stdout, "Reading metadata file: %s\n", filename);
+
+	file_id = H5Fopen(filename, H5F_ACC_RDONLY, H5P_DEFAULT);
+	if (file_id < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to open hdf5 file "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	rc = deserialize_props(poh, file_id, props, cont_type);
+	if (rc != 0) {
+		fprintf(stdout, "failed to deserialize_props\n");
+		D_ERROR("failed to deserialize cont props "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+out:
+	if (file_id > 0)
+		H5Fclose(file_id);
+	return rc;
+}
+
+static int
+deserialize_attrs(hid_t file_id, uint64_t *_num_attrs,
+	    	  char ***_names, void ***_buffers, size_t **_sizes)
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs